client/src/utils/CSSHelp.ts

```diff
@@ -0,0 +1,115 @@
+const CSSTypes = {
+  style: 1,
+  import: 3,
+  media: 4,
+  fontface: 5,
+  keyframes: 7
+};
+
+// @ts-ignore - top-class coding...
+CSSStyleRule.prototype.isDeclaredAfter = function (selector: string) {
```

Comment:
It's angry about us trying to add a property that doesn't match the type def. I played with using `Object.defineProperty` but haven't tested:

```ts
Object.defineProperty(CSSStyleRule.prototype, 'isDeclaredAfter', {
  value: function (selector: string) {
    // Could use some clean up...
    const cssStyleRules = Array.from(
      (this as CSSStyleRule).parentStyleSheet?.cssRules || []
    )?.filter(ele => ele.type === CSSTypes.style) as CSSStyleRule[];
    const previousStyleRule = cssStyleRules.find(
      ele => ele?.selectorText === selector
    );
    if (!previousStyleRule) return false;
    const currPosition = Array.from(
      (this as CSSStyleRule).parentStyleSheet?.cssRules || []
    ).indexOf(this);
    const prevPosition = Array.from(
      previousStyleRule?.parentStyleSheet?.cssRules || []
    ).indexOf(previousStyleRule);
    return currPosition > prevPosition;
  }
});
```

Reply:
If it works, would allow us to remove a bunch of these `ts-ignore`.